pandas/core/frame.py

```diff
@@ -10958,9 +10958,14 @@ def any(  # type: ignore[override]
     ) -> Series:
         # error: Incompatible return value type (got "Union[Series, bool]",
         # expected "Series")
-        return self._logical_func(  # type: ignore[return-value]
+
+        result = self._logical_func(
             "any", nanops.nanany, axis, bool_only, skipna, **kwargs
         )
+        if isinstance(result, (np.bool_, bool)):
+            return result  # type: ignore[return-value]
```

Comment:
Could add `bool | np.bool__` to the return types (but then you might need to add ignores elsewhere in pandas where `any()` is called)

Reply:
Thank you for your suggestion, let me try

Reply:
You probably need to add ignores here:

> pandas/core/reshape/encoding.py:458: error: Item "bool" of "Union[Series, bool_, bool]" has no attribute "any"  [union-attr]
> pandas/core/reshape/encoding.py:461: error: Item "bool_" of "Union[Series, bool_, bool]" has no attribute "idxmax"  [union-attr]
> pandas/core/reshape/encoding.py:461: error: Item "bool" of "Union[Series, bool_, bool]" has no attribute "idxmax"  [union-attr]